stdlib/fun.mli

```diff
@@ -61,3 +63,67 @@ exception Finally_raised of exn
     an unexpected exception or a programming error. As a general rule,
     one should not catch a [Finally_raised] exception except as part of
     a catch-all handler. *)
+
+(** {1:examples Examples}
+
+    {2 Combinators}
+
+    One common mode of usage for {!combinators} is to create and pass anonymous
+    functions as arguments to higher-order functions, without needing to specify
+    [(fun ... -> ...)] explicitly.
+
+    The examples below will be demonstrating this mainly with the {!module:List}
+    module.
+
+    {!val:id}
+    {[
+    # List.init 3 Fun.id;;
+    - : int list = [0; 1; 2]
+
+    # List.find_all Fun.id [false; false; true; false];;
+    - : bool list = [true]
+    ]}
+
+    {!val:const}
+    {[
+    # List.init 3 (Fun.const 0)
+    - : int list = [0; 0; 0]
+    ]}
+    One difference between applying [const (...)] and passing the lambda
+    [(fun _ -> ...)] directly is that the expression denoted [(...)] will be
+    evaluated once, and the function [const] returns will only have access to
+    the result of evaluating [(...)]. So really, [const (...)] is not equivalent
+    to [(fun _ -> ...)], but rather to [let result = ... in (fun _ -> result)].
+
+    To demonstrate this, consider if [(...)] was a call to {!val:Random.bool}:
+
+    for all [n] where [n > 0], [List.init n (Fun.const (Random.bool()))] will
+    have {i exactly two} possible outcomes,
+    - [[true; true; ...; true]] or
+    - [[false; false; ...; false]].
+
+    {!val:flip}
+    {[
+    # List.sort (Fun.flip Int.compare) [5; 3; 9; 0; 1; 6; 8];;
+    - : int list = [9; 8; 6; 5; 3; 1; 0]
+
+    # let subtract = Fun.flip (-) in
+      List.map (subtract 2) [4; 6; 8];;
+    - : int list = [2; 4; 6]
+
+    # List.fold_left (Fun.flip List.cons) [] [1; 2; 3];;
+    - : int list = [3; 2; 1]
+    ]}
+    Note that given a function [(f : a -> b -> c -> d)]:
+    - [(flip f x : a -> c -> d)] and [(x : b)], whereas
+    - [(flip (f x) : c -> b -> d)] and [(x : a)]
+
+    {!val:negate}
```

Comment:
It might work better to have a thematic `predicate` section which could contain example of `negate`, `const true` and `const false`.

Reply:
There's value in having one overarching theme in the examples, I agree, but I feel it'll take a bit of power out of the other examples. WDYT? I'll try to come up with nicer examples to the other three following the predicate theme...